map.go

```diff
@@ -72,6 +72,7 @@ func deepMap(dst, src reflect.Value, visited map[uintptr]*visit, depth int, conf
 	case reflect.Struct:
 		srcMap := src.Interface().(map[string]interface{})
 		for key := range srcMap {
+			config.overwriteWithEmptyValue = true
```

Comment:
if the `continue` or `return` cases are hit below without calling `deepMerge`, this results in mutating `config` and setting `overwriteWithEmptyValue=true` that can affect other calls to `deepMerge`